cpp/src/arrow/flight/sql/odbc/tests/odbc_test_suite.cc

```diff
@@ -23,25 +23,23 @@
 
 // For DSN registration
 #include "arrow/flight/sql/odbc/odbc_impl/config/configuration.h"
+#include "arrow/flight/sql/odbc/odbc_impl/encoding_utils.h"
 #include "arrow/flight/sql/odbc/odbc_impl/odbc_connection.h"
 
 namespace arrow::flight::sql::odbc {
+namespace tests {
 
 void FlightSQLODBCRemoteTestBase::AllocEnvConnHandles(SQLINTEGER odbc_ver) {
   // Allocate an environment handle
-  SQLRETURN ret = SQLAllocEnv(&env);
+  EXPECT_EQ(SQL_SUCCESS, SQLAllocEnv(&env));
 
-  EXPECT_EQ(SQL_SUCCESS, ret);
-
-  ret = SQLSetEnvAttr(env, SQL_ATTR_ODBC_VERSION,
-                      reinterpret_cast<SQLPOINTER>(static_cast<intptr_t>(odbc_ver)), 0);
-
-  EXPECT_EQ(SQL_SUCCESS, ret);
+  EXPECT_EQ(
+      SQL_SUCCESS,
+      SQLSetEnvAttr(env, SQL_ATTR_ODBC_VERSION,
+                    reinterpret_cast<SQLPOINTER>(static_cast<intptr_t>(odbc_ver)), 0));
 
   // Allocate a connection using alloc handle
-  ret = SQLAllocHandle(SQL_HANDLE_DBC, env, &conn);
-
-  EXPECT_EQ(SQL_SUCCESS, ret);
+  EXPECT_EQ(SQL_SUCCESS, SQLAllocHandle(SQL_HANDLE_DBC, env, &conn));
```

Comment:
For disconnect we can have `EXPECT_EQ` to ensure things are freed up, but for the connection phase, should we just use `ASSERT_EQ` to make sure things pass before allowing the tests to continue?